eth/core/state/journal/transient_storage.go

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2023, Berachain Foundation. All rights reserved.
+// Use of this software is govered by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package journal
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"pkg.berachain.dev/polaris/lib/ds"
+	"pkg.berachain.dev/polaris/lib/ds/stack"
+)
+
+// transientState is a representation of EIP-1153 "Transient Storage".
+type transientState map[common.Address]map[common.Hash]common.Hash
+
+// Set sets the transient storage state `value` for `key` at the given `addr`.
+func (t transientState) Set(addr common.Address, key, value common.Hash) {
+	if _, ok := t[addr]; !ok {
+		t[addr] = make(map[common.Hash]common.Hash)
+	}
+	t[addr][key] = value
+}
+
+// Get gets the transient storage state for `key` at the given `addr`.
+func (t transientState) Get(addr common.Address, key common.Hash) common.Hash {
+	val, ok := t[addr]
+	if !ok {
+		return common.Hash{}
+	}
+	return val[key]
+}
+
+// Copy does a deep copy of the transientState.
+func (t transientState) Copy() transientState {
+	storage := make(transientState)
+	for storKey, storVal := range t {
+		valDeepCopy := make(map[common.Hash]common.Hash, len(storVal))
+		for key, val := range storVal {
+			valDeepCopy[key] = val
+		}
+		storage[storKey] = valDeepCopy
+	}
+	return storage
+}
+
+// `transientStorage` is a journal that tracks the transient state.
+type transientStorage struct {
+	ds.Stack[transientState]
+}
+
+// `NewTransientStorage` returns a new `transient` journal.
+func NewTransientStorage() *transientStorage {
```

Comment:
gonna have to add a `nolint:revive` here. Check the pattern used in logs.go, refund.go, etc